src/System.Private.CoreLib/shared/System/Text/Encoding.cs

```diff
@@ -651,10 +652,11 @@ public virtual unsafe int GetByteCount(char* chars, int count)
                       SR.ArgumentOutOfRange_NeedNonNegNum);
 
             char[] arrChar = new char[count];
-            int index;
 
-            for (index = 0; index < count; index++)
-                arrChar[index] = chars[index];
+            fixed (char* pArrChar = arrChar)
```

Comment:
@bartonjs's suggestion from another thread can be improved even further down to a single line:

```cs
char[] arrChar = new Span<char>(chars, count).ToArray();
```

This pattern is likely to have the most optimal performance across all variants because: (a) the bounds checks should be elided due to those same checks having already occurred earlier in the method, and (b) the implementation of `ToArray` can make assumptions which can optimize data copying. For example, the implementation of `ToArray` can (but currently doesn't) skip the "do the source and destination buffers overlap?" check entirely.